src/backend_api.cpp

```diff
@@ -0,0 +1,268 @@
+#include <miopen/backend_api.hpp>
+#include <miopen/miopen.h>
+#include <miopen/errors.hpp>
+#include <new>
+
+extern "C" miopenStatus_t
+miopenBackendCreateDescriptor(miopenBackendDescriptorType_t descriptorType,
+                              miopenBackendDescriptor_t* descriptor)
+{
+    switch(descriptorType)
+    {
+    case MIOPEN_BACKEND_TENSOR_DESCRIPTOR:
+        // TODO: add logging
+        // TODO: get rid of unnecessary throws
+        return miopen::try_(
+            [&] {
+                miopen::backend_api::BackendTensorDescriptor* newDescriptor =
```

Comment:
use `miopen::deref(descriptor)` here. Well, now that I think of it, may be we need to tweak the `miopen::deref` or create a slightly modified version that allows multiple levels of class hierarchy.